core/src/main/java/jenkins/slaves/NoOpComputerLauncher.java

```diff
@@ -23,49 +23,28 @@
  */
 package jenkins.slaves;
 
-import hudson.Extension;
-import hudson.init.Terminator;
-import hudson.model.Computer;
+import hudson.model.TaskListener;
+import hudson.slaves.ComputerLauncher;
+import hudson.slaves.SlaveComputer;
 import java.io.IOException;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-import org.jenkinsci.remoting.protocol.IOHub;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 
 /**
- * Singleton holder of {@link IOHub}
- *
- * @since 2.27
+ * Computer Launcher, which does nothing.
```

Comment:
I do not think this really needs to exist. Allow `SlaveComputer.launcher` to be null and handle that directly when trying to call the launcher.